planemo/shed_lint.py

```diff
@@ -279,3 +320,45 @@ def _validate_categories(categories, realized_repository):
                        "in the category 'Tool Dependency Packages'.")
 
     return msg
+
+
+def _validate_action_url(lint_ctx, action):
+    url = action.text.strip()
+    if url.startswith('http:') or url.startswith('ftp:'):
+        lint_ctx.info(("action resource contains a URL over "
+                       "insecure channels. You may wish to check for "
+                       "a secured (https) download option"))
+
+        insecure_hash_used = any([
+            any([
+                x in url
+                for x in ('#md5=', '#md5#')
+            ]),
+            'md5' in action.attrib
+        ])
+        if insecure_hash_used:
+            lint_ctx.info(("A hash was found, but using "
+                           "a known broken algorithm. Please "
+                           "validate the contents of the download, "
+                           "and generate a sha256 hash"))
+
+        if '#sha256#' in url:
+            lint_ctx.info(("A secure hash was found in the URL, please "
+                           "consider migrating this to using the <action /> "
+                           "property key \"sha256\" to store your hashes "
+                           "as this will not break HTTP spec."))
+
+        if 'sha256' in action.attrib:
```

Comment:
```sha256=``` = ```sha256sum``` (if sha256 is desired, please change it in *lib/tool_shed/galaxy_install/tool_dependencies/recipe/step_handler.py*)